codestodians/komlie-frontend
# [Feature] Prevent spamming of cloudinary bucket with profile images

#### What does this PR do?
- Prevent spamming of cloudinary bucket with profile images

#### Description of Task to be completed?
- refactor the `uploadImage` util to upload images to a signed preset.
- Pass in the `public_id` as a param while uploading to allow for unique resource names

#### How should this be manually tested?
- Check out into the `feature/overwrite-profile-picture-uploads-167314282` branch.
- Add the env variables provided outside this description to your `.env` file
- `nvm use && npm i`
- Start the app with `npm run dev`
- Go to the profile page and upload as many times as possible.
- Check the `avatars` folder in the cloudinary account, only one instance of a particular user's image should be present.

#### Any background context you want to provide?
- Because of the random string generated for the url during unsigned uploads, the images keep piling up and cluttering the cloudinary space. Switching to signed uploads which allow the specification of `publicIds` manually assigned to each image seems to be a useful solution for the cluttering.

#### What are the relevant pivotal tracker stories?
- [#167314282](https://www.pivotaltracker.com/story/show/167314282)
